api/application_msgraph.go

```diff
@@ -52,114 +52,106 @@ func (c *AppClient) AddToUserAgent(extension string) error {
-		result.Response = autorest.Response{Response: resp}
-		err = autorest.NewErrorWithError(err, "provider", "GetApplication", resp, "Failure sending request")
-		return
+		result = ApplicationResult{
+			Response: autorest.Response{Response: resp},
+		}
+		return result, autorest.NewErrorWithError(err, "provider", "GetApplication", resp, "Failure sending request")
 	}
 
 	result, err = c.getApplicationResponder(resp)
 	if err != nil {
-		err = autorest.NewErrorWithError(err, "provider", "GetApplication", resp, "Failure responding to request")
+		return result, autorest.NewErrorWithError(err, "provider", "GetApplication", resp, "Failure responding to request")
 	}
 
-	return
+	return result, nil
 }
 
 // CreateApplication create a new Azure application object.
 func (c *AppClient) CreateApplication(ctx context.Context, displayName string) (result ApplicationResult, err error) {
 	req, err := c.createApplicationPreparer(ctx, displayName)
 	if err != nil {
-		err = autorest.NewErrorWithError(err, "provider", "CreateApplication", nil, "Failure preparing request")
-		return
+		return result, autorest.NewErrorWithError(err, "provider", "CreateApplication", nil, "Failure preparing request")
 	}
 
 	resp, err := c.createApplicationSender(req)
 	if err != nil {
 		result.Response = autorest.Response{Response: resp}
-		err = autorest.NewErrorWithError(err, "provider", "CreateApplication", resp, "Failure sending request")
-		return
+		return result, autorest.NewErrorWithError(err, "provider", "CreateApplication", resp, "Failure sending request")
 	}
 
 	result, err = c.createApplicationResponder(resp)
 	if err != nil {
-		err = autorest.NewErrorWithError(err, "provider", "CreateApplication", resp, "Failure responding to request")
+		return result, autorest.NewErrorWithError(err, "provider", "CreateApplication", resp, "Failure responding to request")
+
 	}
 
-	return
+	return result, nil
 }
 
 // DeleteApplication deletes an Azure application object.
 // This will in turn remove the service principal (but not the role assignments).
-func (c *AppClient) DeleteApplication(ctx context.Context, applicationObjectID string) (result autorest.Response, err error) {
+func (c *AppClient) DeleteApplication(ctx context.Context, applicationObjectID string) (err error) {
 	req, err := c.deleteApplicationPreparer(ctx, applicationObjectID)
 	if err != nil {
-		err = autorest.NewErrorWithError(err, "provider", "DeleteApplication", nil, "Failure preparing request")
-		return
+		return autorest.NewErrorWithError(err, "provider", "DeleteApplication", nil, "Failure preparing request")
 	}
 
 	resp, err := c.deleteApplicationSender(req)
 	if err != nil {
-		result.Response = resp
-		err = autorest.NewErrorWithError(err, "provider", "DeleteApplication", resp, "Failure sending request")
-		return
-	}
-
-	result, err = c.deleteApplicationResponder(resp)
-	if err != nil {
-		err = autorest.NewErrorWithError(err, "provider", "DeleteApplication", resp, "Failure responding to request")
```

Comment:
Not sure if it matters, but the error is no longer "wrapped" this way since we're returning the error directly as-is.

Reply:
Good catch. I've added the wrapping back in here.